src/ol/expr/cpu.js

```diff
@@ -236,17 +251,43 @@ function compileAccessorExpression(expression, context) {
   const name = /** @type {string} */ (nameExpression.value);
   switch (expression.operator) {
     case Ops.Get: {
-      return (context) => context.properties[name];
+      return (context) =>
+        context.properties === UNKNOWN_VALUE
+          ? UNKNOWN_VALUE
+          : context.properties[name];
     }
     case Ops.Var: {
-      return (context) => context.variables[name];
+      return (context) =>
+        context.variables === UNKNOWN_VALUE
+          ? UNKNOWN_VALUE
+          : context.variables[name];
     }
     default: {
       throw new Error(`Unsupported accessor operator ${expression.operator}`);
     }
   }
 }
 
+/**
+ * @param {Array<ExpressionEvaluator>} argEvaluators Argument evaluators
+ * @param {function(Array): ReturnType} evaluator Final evaluator taking in the evaluated args
+ * @return {function(EvaluationContext):ReturnType} the evaluator function; if any arg evaluated to UNKNOWN_VALUE, will return UNKNOWN_VALUE
+ * @template ReturnType
+ */
+function checkForUnknown(argEvaluators, evaluator) {
+  return (context) => {
+    const evaluatedArgs = new Array(argEvaluators.length);
+    for (let i = 0, ii = evaluatedArgs.length; i < ii; i++) {
+      const value = argEvaluators[i](context);
+      if (value === UNKNOWN_VALUE) {
```

Comment:
I'm wondering if we will need to distinguish between unknown and `null`.  For example, a coalesce-type operator like the `'string'` operator returns the first string in an array that may contain `null`.  We wouldn't want the whole expression to be considered unknown in this case.

Basically, I'm wondering if we want unknown be a more reliable sentinel value:
```
const UNKNOWN = {};
```

Reply:
Very good point! I didn't think of that

Reply:
Addressed in #15415